pkg/client/client_test.go

```diff
@@ -1910,7 +1946,344 @@ func TestClose(t *testing.T) {
 
 }
 
+type ClientTestSuiteTrackEvent struct {
+	suite.Suite
+	mockProcessor       *MockProcessor
+	mockDecisionService *MockDecisionService
+}
+
+func (s *ClientTestSuiteTrackEvent) SetupTest() {
+	mockProcessor := new(MockProcessor)
+	mockProcessor.On("ProcessEvent", mock.Anything).Return(true)
+	s.mockProcessor = mockProcessor
+	s.mockDecisionService = new(MockDecisionService)
+}
+
+func (s *ClientTestSuiteTrackEvent) TestTrackWithNotification() {
+
+	expectedUserContext := entities.UserContext{ID: "1212121", Attributes: map[string]interface{}{}}
+
+	isTrackCalled := false
+	onTrack := func(eventKey string, userContext entities.UserContext, eventTags map[string]interface{}, conversionEvent event.ConversionEvent) {
+		isTrackCalled = true
+		s.Equal("sample_conversion", eventKey)
+		s.Equal(expectedUserContext, userContext)
+		s.Equal(*s.mockProcessor.Events[0].Conversion, conversionEvent)
+	}
+
+	client := OptimizelyClient{
```

Comment:
This client initialization is repeated throughout the tests in the suite. We can initialize in the test setup